src/impls/from_iterator.rs

```diff
@@ -20,6 +20,22 @@ where
         counter.update(iterable);
         counter
     }
+
+    /// Create a new `Counter` initialized with the given iterable.
+    /// Allocate with iterator size hints, defaulting to the lowerbound
+    pub fn init_with_capacity<I>(iterable: I) -> Self
+    where
+        I: IntoIterator<Item = T>,
+    {
```

Comment:
I don't think this interface is appropriate. A `with_capacity` method should always accept an explicit capacity from the user. 

I'm considering deprecating the `init` method; it's unidiomatic. In general, the `FromIterator` or `Extend` interfaces should be preferred anyway. 

That said, I would have no objections to moving the capacity guessing optimization here into the `init` implementation.

Reply:
Okay I'll read through the canges and try to understand what is expected here.